core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2767,3 +2767,24 @@ private[spark] class CircularBuffer(sizeInBytes: Int = 10240) extends java.io.Ou
     new String(nonCircularBuffer, StandardCharsets.UTF_8)
   }
 }
+
+
+/**
+ * Factory for URL stream handlers. It relies on 'protocol' to choose the appropriate
+ * UrlStreamHandlerFactory to create URLStreamHandler. Adding new 'if' branches in
+ * 'createURLStreamHandler' like 'hdfsHandler' to support more protocols.
+ */
+private[spark] class SparkUrlStreamHandlerFactory extends URLStreamHandlerFactory {
+  private var hdfsHandler : URLStreamHandler = _
+
+  def createURLStreamHandler(protocol: String): URLStreamHandler = {
+    if (protocol.compareToIgnoreCase("hdfs") == 0) {
+      if (hdfsHandler == null) {
+        hdfsHandler = new FsUrlStreamHandlerFactory().createURLStreamHandler(protocol)
```

Comment:
I think you should call `public FsUrlStreamHandlerFactory(Configuration conf) ` this constructor, and use `Configuration` created by `SparkHadoopUtils`.